### ISSUE TITLE
[Feature]: Safe info about the conversion in the NWB files

### ISSUE THREAD

**Issue by oruebel** [NeurodataWithoutBorders/nwb-guide#939]
### What would you like to see added to the NWB GUIDE?

**Problem** 
It would be useful to be able to:
1. identify if a file was generated with GUIDE. This is useful for usage metrics (e.g., to count the number of files on DANDI that use GUIDE) as well as for degbugging
2. provide information about the conversion process

**Proposed Solution:** 
* NWB has the field ``/general/source_script`` that we could be used to store metadata about the conversion. For the minimal solution, this can be very simple. Maybe a YAML or JSON document with name of the software (i.e. "NWB GUIDE") and version of GUIDE used. We could then build on this approach, to include more information, e.g., to include the full  YAML configuration for the conversion that GUIDE used. 
* Alternatively (or in addition to) we you could append a standard string to the ``/general/notes`` text, e.g., ``This file was generated using NWB GUIDE 1.0``
* It would also be useful to do follow the same approach in NeuroConv (or maybe implement it there)

### Do you have any interest in helping implement the feature?

Yes.

### Code of Conduct

Yes

### Did you confirm this feature was not already reported?

Yes

**Comment by rly** [comment]
> * NWB has the field `/general/source_script` that we could be used to store metadata about the conversion. For the minimal solution, this can be very simple. Maybe a YAML or JSON document with name of the software (i.e. "NWB GUIDE") and version of GUIDE used. We could then build on this approach, to include more information, e.g., to include the full  YAML configuration for the conversion that GUIDE used.

The minimal solution is [already implemented](https://github.com/NeurodataWithoutBorders/nwb-guide/blob/fb3482897c2f042fbf39c7638e67ffc138c6efae/src/pyflask/manageNeuroconv/manage_neuroconv.py#L1005-L1006). Adding information about the conversion process would be nice though.

**Comment by oruebel** [comment]
> The minimal solution is [already implemented](https://github.com/NeurodataWithoutBorders/nwb-guide/blob/fb3482897c2f042fbf39c7638e67ffc138c6efae/src/pyflask/manageNeuroconv/manage_neuroconv.py#L1005-L1006).

Great! Does NeuroConv also do this? 

> Adding information about the conversion process would be nice though.

That can probably done when support for exporting to YAML is possible, because then NeuroConv could just add the YAML script in.

**Comment by rly** [comment]
> Great! Does NeuroConv also do this?

Yes. https://github.com/catalystneuro/neuroconv/blob/60c5e2a9a1736c237ce46f1545abbe0a318070e6/src/neuroconv/basedatainterface.py#L80

**Comment by bendichter** [comment]
I would add to this that there are dandiset metadata fields that we could populate as well
